.golangci.yml

```diff
@@ -15,7 +15,6 @@ linters-settings:
 linters:
   enable:
   - govet
-  - deadcode
```

Comment:
according to https://golangci-lint.run/usage/quick-start/ `deadcode` is enabled by default instead of deprecated. I think with `disable-all` set to true we'll lose `deadcode` check with this change?

Reply:
If you add it again you will get the following message:
`WARN [runner] The linter 'deadcode' is deprecated (since v1.49.0) due to: The owner seems to have abandoned the linter.  Replaced by unused.`

I didn't add unused because it had a lot of findings. But it's something to look at

Reply:
Also if you check https://golangci-lint.run/usage/linters/ you will find the samme message there.

Reply:
```deadcode ⚠ | The owner seems to have abandoned the linter. Replaced by unused.```

Reply:
I see, sounds like it should be replaced by unused

Reply:
Yes, however enabling unused triggers a lot of findings and I didn't see any config to only enable a similar subset as deadcode. I would gladly make follow up prs that enable more linters/explore unused. Similar to what I've done for cluster-api.

Reply:
would you mind push a change with `unused` enabled so there is a record of what the findings are?

Reply:
could be in a separate PR

Reply:
Here is from my local machine

```shell
make go-lint
hack/make-rules/verify/golangci-lint.sh
Ensuring go version.
Install golangci-lint.
Run golangci-lint.
Go version: go version go1.18 linux/amd64
Golangci-lint version: golangci-lint has version v1.49.0 built from (unknown, mod sum: "h1:I8WHOavragDttlLHtSraHn/h39C+R60bEQ5NoGcHQr8=") on (unknown)
prow/cmd/mkpj/main.go:41:2: field `outputPath` is unused (unused)
	outputPath  string
	^
prow/jenkins/jenkins_test.go:127:7: type `Task` is unused (unused)
	type Task struct {
	     ^
gcsweb/cmd/gcsweb/gcsweb.go:523:2: field `isDir` is unused (unused)
	isDir bool
	^
prow/external-plugins/cherrypicker/server_test.go:44:2: field `labels` is unused (unused)
	labels     []github.Label
	^
prow/gerrit/adapter/adapter.go:114:2: field `repoCacheMapMux` is unused (unused)
	repoCacheMapMux         sync.Mutex
	^
prow/plugins/cat/cat.go:97:2: field `keyPath` is unused (unused)
	keyPath string
	^
prow/plugins/slackevents/slackevents_test.go:233:3: field `repoLabels` is unused (unused)
		repoLabels       []string
		^
prow/tide/gerrit_test.go:44:2: field `reviews` is unused (unused)
	reviews int
	^
prow/cmd/pipeline/main.go:48:2: field `buildCluster` is unused (unused)
	buildCluster           string
	^
prow/cmd/prow-controller-manager/main.go:57:2: field `leaderElectionNamespace` is unused (unused)
	leaderElectionNamespace string
	^
prow/cmd/prow-controller-manager/main.go:55:2: field `buildCluster` is unused (unused)
	buildCluster            string
	^
kubetest/conformance/conformance.go:43:2: field `control` is unused (unused)
	control   *process.Control
	^
kubetest/conformance/conformance.go:42:2: field `testArgs` is unused (unused)
	testArgs  *string
	^
kubetest/conformance/conformance.go:40:2: field `e2etest` is unused (unused)
	e2etest   string
	^
kubetest/conformance/conformance.go:38:2: field `kubecfg` is unused (unused)
	kubecfg   string
	^
kubetest/conformance/conformance.go:39:2: field `ginkgo` is unused (unused)
	ginkgo    string
	^
kubetest/conformance/conformance.go:41:2: field `reportdir` is unused (unused)
	reportdir string
	^
prow/plugins/jira/jira_test.go:121:28: func `(*fakeGitHubClient).GetIssue` is unused (unused)
func (f *fakeGitHubClient) GetIssue(org, repo string, number int) (*github.Issue, error) {
                           ^
prow/plugins/jira/jira_test.go:125:28: func `(*fakeGitHubClient).EditIssue` is unused (unused)
func (f *fakeGitHubClient) EditIssue(org, repo string, number int, issue *github.Issue) (*github.Issue, error) {
                           ^
prow/plugins/jira/jira_test.go:113:28: func `(*fakeGitHubClient).EditComment` is unused (unused)
func (f *fakeGitHubClient) EditComment(org, repo string, id int, body string) error {
                           ^
prow/plugins/jira/jira_test.go:109:6: type `fakeGitHubClient` is unused (unused)
type fakeGitHubClient struct {
     ^
prow/spyglass/storageartifact_fetcher.go:225:30: func `(*storageJobSource).canonicalLink` is unused (unused)
func (src *storageJobSource) canonicalLink() string {
                             ^
prow/cmd/generic-autobumper/bumper/bumper.go:46:6: type `fileArrayFlag` is unused (unused)
type fileArrayFlag []string
     ^
prow/cmd/generic-autobumper/bumper/bumper.go:48:26: func `(*fileArrayFlag).String` is unused (unused)
func (af *fileArrayFlag) String() string {
                         ^
prow/cmd/generic-autobumper/bumper/bumper.go:52:26: func `(*fileArrayFlag).Set` is unused (unused)
func (af *fileArrayFlag) Set(value string) error {
                         ^
robots/issue-creator/sources/triage-filer.go:49:2: field `nextSync` is unused (unused)
	nextSync    time.Time
	^
prow/spyglass/lenses/html/html_test.go:38:2: field `sizeLimit` is unused (unused)
	sizeLimit int64
	^
prow/plugins/goose/goose.go:94:2: field `keyPath` is unused (unused)
	keyPath string
	^
prow/cmd/crier/main.go:61:2: field `k8sGCSWorkers` is unused (unused)
	k8sGCSWorkers         int
	^
prow/cmd/sinker/main.go:185:2: field `cancel` is unused (unused)
	cancel        context.CancelFunc
	^
robots/commenter/main.go:89:2: field `asc` is unused (unused)
	asc             bool
	^
robots/commenter/main.go:97:2: field `sort` is unused (unused)
	sort            string
	^
prow/kube/ratelimiter.go:45:24: func `(*fakeLimiter).Done` is unused (unused)
func (fl *fakeLimiter) Done(interface{})   {}
                       ^
prow/kube/ratelimiter.go:58:24: func `(*fakeLimiter).NumRequeues` is unused (unused)
func (fl *fakeLimiter) NumRequeues(item interface{}) int {
                       ^
prow/kube/ratelimiter.go:41:24: func `(*fakeLimiter).ShutDown` is unused (unused)
func (fl *fakeLimiter) ShutDown() {}
                       ^
prow/kube/ratelimiter.go:42:24: func `(*fakeLimiter).Get` is unused (unused)
func (fl *fakeLimiter) Get() (interface{}, bool) {
                       ^
prow/kube/ratelimiter.go:46:24: func `(*fakeLimiter).Forget` is unused (unused)
func (fl *fakeLimiter) Forget(interface{}) {}
                       ^
prow/kube/ratelimiter.go:50:24: func `(*fakeLimiter).Add` is unused (unused)
func (fl *fakeLimiter) Add(a interface{}) {
                       ^
prow/kube/ratelimiter.go:47:24: func `(*fakeLimiter).AddRateLimited` is unused (unused)
func (fl *fakeLimiter) AddRateLimited(a interface{}) {
                       ^
prow/kube/ratelimiter.go:53:24: func `(*fakeLimiter).AddAfter` is unused (unused)
func (fl *fakeLimiter) AddAfter(a interface{}, d time.Duration) {
                       ^
prow/kube/ratelimiter.go:56:24: func `(*fakeLimiter).Len` is unused (unused)
func (fl *fakeLimiter) Len() {
                       ^
prow/kube/ratelimiter.go:37:6: type `fakeLimiter` is unused (unused)
type fakeLimiter struct {
     ^
kubetest/extract_k8s.go:134:26: func `extractStrategy.name` is unused (unused)
func (e extractStrategy) name() string {
                         ^
kubetest/main.go:119:2: field `sshUser` is unused (unused)
	sshUser                 string
	^
kubetest/kops.go:165:2: field `multipleZones` is unused (unused)
	multipleZones bool
	^
kubetest/main.go:108:2: field `outputDir` is unused (unused)
	outputDir               string
	^
greenhouse/diskcache/cache.go:42:2: field `logger` is unused (unused)
	logger   *logrus.Entry
	^

```

Reply:
Unfortunately the link to golangci-lint's unused configuration seems to be dead:
https://golangci-lint.run/usage/linters/#unused

compare this to e g staticcheck
https://golangci-lint.run/usage/linters/#staticcheck

Reply:
This are great findings!

@alvaroaleman , could you take a look at this list and check whether there is any exposed functions that are consumed by Openshift?

@oscr , now that think about it more I might prefer keeping `deadcode` until either it's broken or we figured out how to use `unused`

Reply:
@chaodaiG No problem. It's been added again